The job **`x86_64-gnu-tools`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/20135850333/job/57788919938) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/57788919938) [(plain)](https://github.com/rust-lang/rust/commit/04aec44e860cca38a0cc627a9ad26c7774d44d9a/checks/57788919938/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
tests/ui/size_of_in_element_count/functions.rs ... ok
tests/ui/crashes/third-party/conf_allowlisted.rs ... ok

FAILED TEST: tests/ui/crashes/ice-6255.rs
command: CLIPPY_CONF_DIR="tests" RUSTC_ICE="0" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/release/clippy-driver" "--error-format=json" "--emit=metadata" "-Aunused" "-Ainternal_features" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Dwarnings" "-Ldependency=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/release/deps" "--sysroot=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/0/tests/ui/crashes" "tests/ui/crashes/ice-6255.rs" "--extern" "futures=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/debug/deps/libfutures-56bbad2987f1a076.rlib" "--extern" "futures=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/debug/deps/libfutures-56bbad2987f1a076.rmeta" "--extern" "itertools=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/debug/deps/libitertools-49215d6c78da5485.rlib" "--extern" "itertools=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/debug/deps/libitertools-49215d6c78da5485.rmeta" "--extern" "libc=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/debug/deps/liblibc-9fd0632ab0adaa16.rlib" "--extern" "libc=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/debug/deps/liblibc-9fd0632ab0adaa16.rmeta" "--extern" "parking_lot=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/debug/deps/libparking_lot-cc8a335fe9146be3.rlib" "--extern" "parking_lot=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/debug/deps/libparking_lot-cc8a335fe9146be3.rmeta" "--extern" "quote=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/debug/deps/libquote-aa12901b2adae6fc.rlib" "--extern" "quote=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/debug/deps/libquote-aa12901b2adae6fc.rmeta" "--extern" "regex=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/debug/deps/libregex-6ecd666cd642c024.rlib" "--extern" "regex=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/debug/deps/libregex-6ecd666cd642c024.rmeta" "--extern" "serde=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/debug/deps/libserde-6629799eafe4f0d4.rlib" "--extern" "serde=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/debug/deps/libserde-6629799eafe4f0d4.rmeta" "--extern" "syn=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/debug/deps/libsyn-b0c36228d90d575f.rlib" "--extern" "syn=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/debug/deps/libsyn-b0c36228d90d575f.rmeta" "--extern" "tokio=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/debug/deps/libtokio-6ddc89563990f085.rlib" "--extern" "tokio=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/debug/deps/libtokio-6ddc89563990f085.rmeta" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/debug/deps" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/debug/deps" "--edition" "2024"

error: actual output differed from expected
Execute `./x test src/tools/clippy --bless` to update `tests/ui/crashes/ice-6255.stderr` to the actual output
--- tests/ui/crashes/ice-6255.stderr
+++ <stderr output>
---
-   |
-LL |         extern crate std as core;
-   |         ^^^^^^^^^^^^^^^^^^^^^^^^^
-...
-LL | define_other_core!();
-   | -------------------- in this macro invocation
-   = help: use `crate::core` to refer to this crate unambiguously
-   = note: this error originates in the macro `define_other_core` (in Nightly builds, run with -Z macro-backtrace for more info)
+error: aborting due to 1 previous error
 
---
   |
LL |         extern crate std as core;
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^
...
LL | define_other_core!();
   | -------------------- in this macro invocation
   |
   = note: this error originates in the macro `define_other_core` (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to 1 previous error



error: ``core` is ambiguous` not found in diagnostics on line 12
##[error]  --> tests/ui/crashes/ice-6255.rs:12:32
   |
12 |     core::panic!(); //~ ERROR: `core` is ambiguous
   |                                ^^^^^^^^^^^^^^^^^^^ expected because of this pattern
   |
---
   |
LL |         extern crate std as core;
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^
...
LL | define_other_core!();
   | -------------------- in this macro invocation
   |
   = note: this error originates in the macro `define_other_core` (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to 1 previous error
```

</details>

--- a/compiler/rustc_resolve/src/ident.rs
+++ b/compiler/rustc_resolve/src/ident.rs
@@ -771,36 +771,39 @@ impl<'ra, 'tcx> Resolver<'ra, 'tcx> {
         } else {
             None
         };
-        // Skip ambiguity errors for extern flag bindings "overridden"
-        // by extern item bindings.
-        // FIXME: Remove with lang team approval.
-        let issue_145575_hack = Some(binding) == extern_prelude_flag_binding
-            && extern_prelude_item_binding.is_some()
-            && extern_prelude_item_binding != Some(innermost_binding);
-        if let Some(kind) = ambiguity_error_kind
-            && !issue_145575_hack
-        {
-            let misc = |f: Flags| {
-                if f.contains(Flags::MISC_SUGGEST_CRATE) {
-                    AmbiguityErrorMisc::SuggestCrate
-                } else if f.contains(Flags::MISC_SUGGEST_SELF) {
-                    AmbiguityErrorMisc::SuggestSelf
-                } else if f.contains(Flags::MISC_FROM_PRELUDE) {
-                    AmbiguityErrorMisc::FromPrelude
-                } else {
-                    AmbiguityErrorMisc::None
-                }
-            };
-            self.ambiguity_errors.push(AmbiguityError {
-                kind,
-                ident: orig_ident,
-                b1: innermost_binding,
-                b2: binding,
-                warning: false,
-                misc1: misc(innermost_flags),
-                misc2: misc(flags),
-            });
-            return true;
+        if let Some(kind) = ambiguity_error_kind {
+            // Skip ambiguity errors for extern flag bindings "overridden"
+            // by extern item bindings.
+            // FIXME: Remove with lang team approval.
+            let issue_145575_hack = Some(binding) == extern_prelude_flag_binding
+                && extern_prelude_item_binding.is_some()
+                && extern_prelude_item_binding != Some(innermost_binding);
+
+            if issue_145575_hack {
+                self.issue_145575_hack_applied = true;
+            } else {
+                let misc = |f: Flags| {
+                    if f.contains(Flags::MISC_SUGGEST_CRATE) {
+                        AmbiguityErrorMisc::SuggestCrate
+                    } else if f.contains(Flags::MISC_SUGGEST_SELF) {
+                        AmbiguityErrorMisc::SuggestSelf
+                    } else if f.contains(Flags::MISC_FROM_PRELUDE) {
+                        AmbiguityErrorMisc::FromPrelude
+                    } else {
+                        AmbiguityErrorMisc::None
+                    }
+                };
+                self.ambiguity_errors.push(AmbiguityError {
+                    kind,
+                    ident: orig_ident,
+                    b1: innermost_binding,
+                    b2: binding,
+                    warning: false,
+                    misc1: misc(innermost_flags),
+                    misc2: misc(flags),
+                });
+                return true;
+            }
         }
 
         false

--- a/compiler/rustc_resolve/src/imports.rs
+++ b/compiler/rustc_resolve/src/imports.rs
@@ -1170,7 +1170,7 @@ impl<'ra, 'tcx> Resolver<'ra, 'tcx> {
                             return;
                         }
                         if let Some(initial_res) = initial_res {
-                            if res != initial_res {
+                            if res != initial_res && !this.issue_145575_hack_applied {
                                 span_bug!(import.span, "inconsistent resolution for an import");
                             }
                         } else if this.privacy_errors.is_empty() {

--- a/compiler/rustc_resolve/src/lib.rs
+++ b/compiler/rustc_resolve/src/lib.rs
@@ -1186,6 +1186,7 @@ pub struct Resolver<'ra, 'tcx> {
     privacy_errors: Vec<PrivacyError<'ra>> = Vec::new(),
     /// Ambiguity errors are delayed for deduplication.
     ambiguity_errors: Vec<AmbiguityError<'ra>> = Vec::new(),
+    issue_145575_hack_applied: bool = false,
     /// `use` injections are delayed for better placement and deduplication.
     use_injections: Vec<UseError<'tcx>> = Vec::new(),
     /// Crate-local macro expanded `macro_export` referred to by a module-relative path.

--- a/tests/ui/resolve/ice-inconsistent-resolution-149821.rs
+++ b/tests/ui/resolve/ice-inconsistent-resolution-149821.rs
@@ -1,0 +1,17 @@
+//@ edition: 2024
+
+mod m {
+    use crate::*;
+    use core;
+}
+
+macro_rules! define_other_core {
+    () => {
+        extern crate std as core;
+        //~^ ERROR macro-expanded `extern crate` items cannot shadow names passed with `--extern`
+    };
+}
+
+define_other_core! {}
+
+fn main() {}

--- a/tests/ui/resolve/ice-inconsistent-resolution-149821.stderr
+++ b/tests/ui/resolve/ice-inconsistent-resolution-149821.stderr
@@ -1,0 +1,13 @@
+error: macro-expanded `extern crate` items cannot shadow names passed with `--extern`
+  --> $DIR/ice-inconsistent-resolution-149821.rs:10:9
+   |
+LL |         extern crate std as core;
+   |         ^^^^^^^^^^^^^^^^^^^^^^^^^
+...
+LL | define_other_core! {}
+   | --------------------- in this macro invocation
+   |
+   = note: this error originates in the macro `define_other_core` (in Nightly builds, run with -Z macro-backtrace for more info)
+
+error: aborting due to 1 previous error
+

